models/favourite.models.js

```diff
@@ -3,98 +3,104 @@ const { fetchUserByUserId } = require("./user.models");
 const { fetchClothByClothesId } = require("./clothes.models");
 
 async function fetchUserFavouriteClothes(user_id) {
-  await fetchUserByUserId(user_id);
+	await fetchUserByUserId(user_id);
 
-  try {
-    const selectUserFavouriteClothes = `
+	try {
+		const selectUserFavouriteClothes = `
         SELECT c.*, f.favourite_id  
         FROM clothes c 
         INNER JOIN favourites f 
         ON c.clothes_id = f.clothes_id 
         WHERE f.uid = $1;
         `;
 
-    const { rows: userFavouriteClothes } = await db.query(
-      selectUserFavouriteClothes,
-      [user_id]
-    );
+		const { rows: userFavouriteClothes } = await db.query(selectUserFavouriteClothes, [user_id]);
 
-    return userFavouriteClothes;
-  } catch (error) {
-    throw error;
-  }
+		return userFavouriteClothes;
+	} catch (error) {
+		throw error;
+	}
 }
 
 async function addFavourite(user_id, newFavourite) {
-  try {
-    if (!newFavourite.clothes_id) {
-      throw {
-        status: 400,
-        msg: "Bad request!",
-      };
-    }
-    if (typeof newFavourite.clothes_id !== "number") {
-      throw {
-        status: 400,
-        msg: `Favourite clothes_id ${newFavourite.clothes_id} should have a number type`,
-      };
-    }
-
-    await fetchUserByUserId(user_id);
-    await fetchClothByClothesId(newFavourite.clothes_id);
-
-    const favouriteData = [newFavourite.clothes_id, user_id];
-
-    const insertUserFavourite = `
+	try {
+		if (!newFavourite.clothes_id) {
+			throw {
+				status: 400,
+				msg: "Bad request!",
+			};
+		}
+		if (typeof newFavourite.clothes_id !== "number") {
+			throw {
+				status: 400,
+				msg: `Favourite clothes_id ${newFavourite.clothes_id} should have a number type`,
+			};
+		}
+
+		await fetchUserByUserId(user_id);
+		await fetchClothByClothesId(newFavourite.clothes_id);
+		const favouriteClothes = await fetchUserFavouriteClothes(user_id);
+		const filterFavourite = favouriteClothes.filter(
+			(currFav) => currFav.clothes_id === newFavourite.clothes_id
+		);
```

Comment:
We can apply the same filtering at the frontend and avoid sending additional requests to the backend, can't we?